### ISSUE TITLE
Sequential generation error

### ISSUE THREAD

**Issue by carpedm20** [carpedm20/pixel-rnn-tensorflow#1]
Current implementation successfully works as a autoencoder but it can't generate from the beginning, from a blank image. This error should be related to masking.

**Comment by carpedm20** [comment]
![image](https://cloud.githubusercontent.com/assets/3346407/17057498/477da3d2-5055-11e6-992d-783b947ad066.png)

(`Up`: ground truth, `Down`: prediction)

Fix with 4c459c11394657c60b12ca0371aadded5e1f208b

**Comment by carpedm20** [comment]
![image](https://cloud.githubusercontent.com/assets/3346407/17074835/dba72ac6-50bc-11e6-89e8-e03f75d8f23c.png)

Even the one-shot recovering of the original image can be learned quickly as shown in previous comment, the sequential sampling of an image is still not working like [igul222/pixel_rnn](https://github.com/igul222/pixel_rnn).

**Comment by carpedm20** [comment]
Update.

![generation_2016_08_01_16_40_28](https://cloud.githubusercontent.com/assets/3346407/17314133/13672276-589d-11e6-9fc9-94bcd1774783.jpg)

Trained model with `python main.py --data=mnist --model=pixel_cnn`
